.travis.yml

```diff
@@ -10,20 +10,44 @@ notifications:
       - ros-contributions@amazon.com
       - travis-build@platform-notifications.robomaker.aws.a2z.com
 env:
+  global:
+    - GITHUB_ACCOUNT=aws-robotics
+    # uncomment PACKAGE_NAME to build unit tests - REQUIRED if you'd like to run unit tests using colcon
   matrix:
-    - ROS_DISTRO="kinetic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
-    - ROS_DISTRO="kinetic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
-    - ROS_DISTRO="lunar"   ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
-    - ROS_DISTRO="lunar"   ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
-    - ROS_DISTRO="melodic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
-    - ROS_DISTRO="melodic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
+    # define DOWNSTREAM_GITHUB_REPO=name-of-your-repo if you'd like to trigger downstream repos' travis builds
+    - ROS_DISTRO=kinetic ROS_VERSION=1
+    - ROS_DISTRO=lunar   ROS_VERSION=1
+    - ROS_DISTRO=melodic ROS_VERSION=1
+    - ROS_DISTRO=bouncy  ROS_VERSION=2
+    - ROS_DISTRO=crystal ROS_VERSION=2
 matrix:
   allow_failures:
-    - env: ROS_DISTRO="lunar"   ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
-    - env: ROS_DISTRO="lunar"   ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
-    - env: ROS_DISTRO="melodic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
-    - env: ROS_DISTRO="melodic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
+    - env: ROS_DISTRO=lunar   ROS_VERSION=1
+    - env: ROS_DISTRO=melodic ROS_VERSION=1
+    - env: ROS_DISTRO=bouncy  ROS_VERSION=2
+    - env: ROS_DISTRO=crystal ROS_VERSION=2
 install:
-  - git clone https://github.com/ros-industrial/industrial_ci.git .ros_ci
+  - git clone https://github.com/xabxx/travis-scripts.git .ros_ci
 script:
-  - .ros_ci/travis.sh
+  # make travis scripts executable
```

Comment:
There's a lot of boilerplate stuff from line 32 onwards that we're going to be repeating for every travis.yml file, can these lines be moved into a script that'd be part of the travis-scripts package? (see how industrial-ci does it)